src/core/tsi/ssl_transport_security.c

```diff
@@ -49,6 +49,8 @@
 #include <openssl/x509.h>
 #include <openssl/x509v3.h>
 
+#include "src/core/support/env.h"
```

Comment:
Agree that an env variable is suboptimal, and with the concern about being overloaded ;).

I don't have a good understanding of all of the constraints that are on this code, but I can take a stab at the enforce_mutual_auth approach. No promises, per the above comment.

Regarding the proposed approach, it won't scale to other kinds of change to the underlying TLS connection- not sure if you consider that a plus or a minus. I prefer it over the alternative simply because it seems to be something that could be implemented and in on a short timescale.